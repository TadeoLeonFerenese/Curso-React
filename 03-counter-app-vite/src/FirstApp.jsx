import PropTypes from "prop-types";
//Los  functionals components como este si se puede se dejan por fuera de la funcion principal por que consumen espacio en memoria cuando re re renderiza la apk
// const getSaludo = () => {
//   return "Hola Mundo :) ";
// };

// const newMessage = "Tadeo!!";
// si quiero mandar un objeto como child  si o si tenes que usar "JSON.stringify()"

export const FirstApp = ({ title, subTitle, name }) => {
  return (
    //esto es un fragmento, sirve para utilizar dos elementos sin tener la necesidad de poner un div
    <>
      <h1 data-testid="test-title">{title}</h1>
      {/* <h1>{getSaludo()}</h1> */}
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

// las proptypes sirven para especificar que tipo de props estas enviando, si es string, number, bolean
// El "isRequired" sirve para que sea si o si  enviado o salta un warning d e error
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.number.isRequired,
};

FirstApp.defaultProps = {
  title: " No hay titulo ",
  subTitle: " No hay subTitle ",
  name: "Tadeo Leon",
};
