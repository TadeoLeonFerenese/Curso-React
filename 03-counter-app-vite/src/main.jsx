import React from "react";
import reactDom from "react-dom/client";
import { HelloWorldApp } from "./HelloWorldApp";
import { FirstApp } from "./FirstApp";
import "./styles.css";
import { CounterApp } from "./CounterApp";

reactDom.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <HelloWorldApp /> */}
    <FirstApp title="Hola, soy Goku" subTitle={123123} />
    <CounterApp value={3640} />
  </React.StrictMode>
);
