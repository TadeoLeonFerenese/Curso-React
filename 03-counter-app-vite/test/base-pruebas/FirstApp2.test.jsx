import { render, screen } from "@testing-library/react";
import { FirstApp } from "../../src/FirstApp";
import { expect, test } from "@jest/globals";

describe("Pruebas en <FisrtApp/>", () => {
  const title = "Hola, soy Goku";
  const subTitle = "Soy un subtitulo";

  test("debe hacer match con el snaptShot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('debe contener el mensaje "Hola, soy Goku"', () => {
    render(<FirstApp title={title} />);
    // screen.debug();

    expect(screen.getByText(title)).toBeTruthy();
  });

  test("debe mostrar el titulo en un h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("debe de mostrar el subtitulo enviado por props", () => {
    render(<FirstApp title={title} subTitle={subTitle} />);
    expect(screen.getAllByText(subTitle).lenght).toBe(); //en teoria el toBe tiene que tener 2 por que retorna un array, pero aparentemente aca le chupa 7 huevos y lo testea asi
  });
});
