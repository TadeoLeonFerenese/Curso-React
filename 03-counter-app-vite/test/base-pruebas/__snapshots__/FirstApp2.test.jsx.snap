// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FisrtApp/> debe hacer match con el snaptShot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <p>
     No hay subTitle 
  </p>
  <p>
     No hay subTitle 
  </p>
  <p>
    Tadeo Leon
  </p>
</div>
`;
